### DISCUSSION TITLE
Fields reappear after removing from blueprint

### DISCUSSION THREAD

**Discussion by kilsicles** [statamic/cms#10644] [Troubleshooting] [answered]
I have this issue for some fields only, but could not really figure out the scheme yet.

Steps to reproduce:
1. Add field to blueprint (e.g. Field "date" of type Date) + save blueprint
2. Remove field from blueprint + save blueprint

Field is still there, sometimes it also gets recreated in other sections of the blueprint.

Notes: 
- When removing the field (via UI) the .yaml file is untouched - the fields are still listed there. When removing them manually, the fields are recreated in the .yaml file after saving the blueprint.
- Tried to remove the cache (`php artisan cache:clear` and `php please stache:clear`), even deleted the data in `framework/cache/`
- As one "non-deletable" field in my case is marked as "required", I can't keep on working at the moment.

Thanks for every idea!

**Comment by duncanmcclean** [comment]
If you've enabled "Publish Dates" on the collection, it'll add back a Date field if you remove it since it's required for that feature to work.

**Reply by kilsicles** [reply]
I face the same issue for e.g. 2 fields of type "Taxonomies" (In example below: "Topic",  "News").

Before saving the blueprint:
<img width="201" alt="image" src="https://github.com/user-attachments/assets/2465c891-0252-4ce1-915b-0ad6f7c47410">
Results in:
<img width="819" alt="image" src="https://github.com/user-attachments/assets/4f7b49f5-627a-422c-a2f1-07b7a3fd015b">
Also after saving the blueprint (Section Resources only):
<img width="149" alt="image" src="https://github.com/user-attachments/assets/1a23e06d-75ee-4d08-838d-f9c05ae31ddb">

**Reply by duncanmcclean** [reply, marked as answer]
Taxonomies work the same way. If you have taxonomies configured on your collection, it'll force those fields onto the blueprint. It's intentional.

**Reply by kilsicles** [reply]
Ok, thanks! Got it (:

Why though??

**Reply by duncanmcclean** [reply]
The fields need to exist in order for the date/taxonomy features to work. If you don't need them, you can disable the features in the collection's settings.
